executor/aggregate.go

```diff
@@ -816,6 +853,117 @@ func (e *HashAggExec) getPartialResults(groupKey string) []aggfuncs.PartialResul
 	return partialResults
 }
 
+func (e *HashAggExec) initRuntimeStats() {
+	if e.runtimeStats != nil && e.stats == nil {
+		e.stats = &HashAggRuntimeStats{
+			PartialNum:        e.ctx.GetSessionVars().HashAggPartialConcurrency(),
+			FinalNum:          e.ctx.GetSessionVars().HashAggFinalConcurrency(),
+			PartialStats:      &AggWorkerStat{},
+			FinalStats:        &AggWorkerStat{},
+			PartialWorkerTime: make([]time.Duration, e.ctx.GetSessionVars().HashAggPartialConcurrency()),
+			FinalWorkerTime:   make([]time.Duration, e.ctx.GetSessionVars().HashAggFinalConcurrency()),
+		}
+		e.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl.RegisterStats(e.id, e.stats)
+	}
+}
+
+// HashAggRuntimeStats record the HashAggExec runtime stat
+type HashAggRuntimeStats struct {
+	PartialNum int
+	FinalNum   int
+
+	PartialStats *AggWorkerStat
```

Comment:
```suggestion
	PartialStats AggWorkerStat
```

No need to be a point, since it should never be nil, and a little bit hard to handle point when `Merge` and `Clone`.